R/pkg/R/utils.R

```diff
@@ -851,3 +851,12 @@ rbindRaws <- function(inputData){
   out[!rawcolumns] <- lapply(out[!rawcolumns], unlist)
   out
 }
+
+# Get basename without extension from URL
+basenameSansExtFromUrl <- function(url) {
```

Comment:
My concern was to bring in another dependencies just for this (it's in the tools)

The regex was in fact copy-paste from file_path_sans_ext (hence the name) except for the compression part which is what you are referring to.
I could copy that over as well. Would you prefer `compression` be TRUE (default is FALSE) to remove `.gz`?

```
> library(tools)
> file_path_sans_ext
function (x, compression = FALSE)
{
    if (compression)
        x <- sub("[.](gz|bz2|xz)$", "", x)
    sub("([^.]+)\\.[[:alnum:]]+$", "\\1", x)
}
```